### PR TITLE
Migrate crypt32 APIs

### PR DESCRIPTION
**Pull request by skdltmxn**

### CODE REVIEW

**Comment by Susurrus** [comment]
Looked over everything but the `wincrypt.rs` changes and those LGTM.

File: src/um/wincrypt.rs [hunk 1]
```diff
@@ -553,6 +592,266 @@ STRUCT!{struct CMS_DH_KEY_INFO {
     pReserved: *mut c_void,
 }}
 pub type PCMS_DH_KEY_INFO = *mut CMS_DH_KEY_INFO;
+extern "system" {
+    pub fn CryptAcquireContextA(
+        phProv: *mut HCRYPTPROV,
+        szContainer: LPCSTR,
+        szProvider: LPCSTR,
+        dwProvType: DWORD,
+        dwFlags: DWORD
```
Comment (retep998) [inline src/um/wincrypt.rs]: Trailing commas!

**Comment by skdltmxn** [comment]
Fixed :)

File: src/um/wincrypt.rs [hunk 2]
```diff
@@ -1334,279 +1745,146 @@ pub const szOID_PROTECTED_PROCESS_SIGNER: &'static str = "1.3.6.1.4.1.311.10.3.2
...
+    RestrictedKeyUsage: CRYPT_BIT_BLOB,
+}}
+pub type PCERT_KEY_USAGE_RESTRICTION_INFO = *mut CERT_KEY_USAGE_RESTRICTION_INFO;
+STRUCT!{struct CERT_OTHER_NAME {
+    pszObjId: LPSTR,
+    Value: CRYPT_OBJID_BLOB,
+}}
+pub type PCERT_OTHER_NAME = *mut CERT_OTHER_NAME;
+UNION2!{union CERT_ALT_NAME_ENTRY_u {
+    [u32; 2] [u32; 5],
```
Comment (retep998) [inline src/um/wincrypt.rs]: According to local testing, this should actually be `[u32; 2] [u64; 2]` or simply `[usize; 2]`.

Comment (skdltmxn) [inline src/um/wincrypt.rs]: You're right, my bad :(
Fixed!

**Pull request merged**